dvc/repo/experiments/show.py

```diff
@@ -368,6 +416,7 @@ def show(
     found_revs.update(
         iter_revs(repo.scm, revs, num, all_branches, all_tags, all_commits)
     )
+    branch_names = get_branch_names(repo.scm, found_revs)
```

Comment:
Move the branch_names look up can make sure the live experiment's revision can be found in the following list of revisions, in previous because it only occurred at the end of the who `exp show` the latest `revision` of `exp-****` reference may not be found in the returning list of revisions.